### PR TITLE
Airport Challenge - Solution

### PR DESCRIPTION
**Pull request by contimarta**
I have used 9 tests to develop my code and I've made it as loosely coupled as I've been able to. My domain models can be found in the README file. You can also find an aiportRunner file where I've used most of the airport class methods under different circumstances.

### CODE REVIEW

File: src/airport.js [hunk 1]
```diff
@@ -1,26 +1,28 @@
 
 class Airport {
   airportPlanes;
+  capacity;
   constructor (typeOfTransport, capacity=20, airportPlanes=[]){
```
Comment (contimarta) [inline src/airport.js:5]: I think that there is a better way to create this constructor but I'm not sure how. In this case, for instance, if I want to initialise an airport object with several planes but without changing the default capacity, I still need to enter a capacity as an argument. Maybe changing the order would help: constructor (typeOfTransport, airportPlanes=[], capacity=20){

File: src/airport.js [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
     this.capacity=capacity;
     this.airportPlanes = airportPlanes
     this.typeOfTransport = typeOfTransport
   }
 
   landPlane(plane){
-    if ( plane.constructor.name == this.typeOfTransport.name) {
+    let indexPlaneRemove= this.airportPlanes.findIndex(element => element.id === plane.id)
+    if ( plane.constructor.name == this.typeOfTransport.name && indexPlaneRemove===-1) {
```
Comment (contimarta) [inline src/airport.js]: I've had to use the property "constructor.name" and "name" respectively given that I've decoupled the class airport from the class plane, and I had to make sure the object that wants to land is a plane. I'm not sure this is the best way to do it, though

File: README.md [hunk 3]
```diff
@@ -95,8 +95,8 @@ Your code should defend against [edge cases](http://programmers.stackexchange.co
 | Airport | airportPlanes[@Plane]     | landPlane(@Plane) | @Void   |
 | Plane   | id @String                |                   |         |
 
-// Test 1 - Checks that landPlane instructs airport to land a Plane
-// Test 2 - Tests that method landPlane only works with a plane
+**Test 1** - Checks that landPlane instructs airport to land a Plane
+**Test 2** - Tests that method landPlane only works with a plane
```
Comment (contimarta) [inline README.md:100]: I was unsure whether the method landPlane should, apart from instructing the airport to land a plane, also add a plane into the array airportPlanes. If that's the case, which is what I've done, does the method fulfil the single responsibility principle? A single method is doing two things: adding a plane to the array, and instructing the airport to land a plane.

File: src/airport.js [hunk 4]
```diff
@@ -33,12 +34,15 @@ class Airport {
     
     
   }
+
   isAtCapacity(){
```
Comment (contimarta) [inline src/airport.js:38]: I would like to use the method isAtCapacity in the landPlane method instead of checking again the condition "this.airportPlanes.length>= this.capacity". This would require that the isAtCapacity method returns a "true" or "false" boolean instead of a string, and maybe use a console.log to display that the airport is full when that's the case.
